code/Native/SerialBuffer.cs

```diff
@@ -197,7 +197,7 @@ bool ISerialBufferStreamData.WaitForRead(int timeout)
                 // Monitoring thread died. No point waiting any longer.
                 return false;
             }
-            throw new ApplicationException("Unexpected code flow");
+            throw new Exception("Unexpected code flow");
```

Comment:
Figured that one out. created a new exception InternalApplicationException which I raise now in the code on the branch feature/nestandard15